cohttp-eio/src/server.mli

```diff
@@ -0,0 +1,3 @@
+include Cohttp.Server.S with module IO = Io.IO and type body = Body.t
+
+val run : #Eio.Net.listening_socket -> t -> unit
```

Comment:
I suggest adding all optional arguments that can be passed to `Eio.Net.run_server`:

```
  ?max_connections:int ->
  ?additional_domains:(#Domain_manager.t * int) ->
  ?stop:'a Promise.t ->
  on_error:(exn -> unit) ->
```
These can be passed to directly to `Eio.Net.run_server`.

Reply:
Another option would be to expose the connection handler function, so that callers can call `run_server` themselves, e.g.

```ocaml
Eio.Net.run_server socket (Cohttp_eio.handle_connection http) ...
```

Reply:
This is, I think, already the case: the "actual" API from `cohttp` we implement is `  val callback : t -> IO.conn -> IO.ic -> IO.oc -> unit IO.t`, which does not concern itself with connection handling. `Server.run` is just a convenience specific to cohttp-eio for the most common case.

- I'll still expose more options as this can hardly hurt.